src/Swarm/Game/Step.hs

```diff
@@ -1313,12 +1359,19 @@ evalCmp c v1 v2 = decideCmp c $ compareValues v1 v2
 -- | Compare two values, returning an 'Ordering' if they can be
 --   compared, or @Nothing@ if they cannot.
 compareValues :: Has (Throw Exn) sig m => Value -> Value -> m Ordering
-compareValues = \v1 -> case v1 of
+compareValues v1 = case v1 of
   VUnit -> \case VUnit -> return EQ; v2 -> incompatCmp VUnit v2
-  VInt n1 -> \case VInt n2 -> return (compare n1 n2); v2 -> incompatCmp v1 v2
   VString t1 -> \case VString t2 -> return (compare t1 t2); v2 -> incompatCmp v1 v2
   VDir d1 -> \case VDir d2 -> return (compare d1 d2); v2 -> incompatCmp v1 v2
   VBool b1 -> \case VBool b2 -> return (compare b1 b2); v2 -> incompatCmp v1 v2
+  VInt n1 -> \case
+    VInt n2 -> return (compare n1 n2)
+    VAsync (GAsync n2 _) -> return $ compare n1 (fromIntegral n2)
```

Comment:
I don't understand this at all.  Why would we compare a `VInt` against the number in a `GAsync`?

Reply:
Fixed :slightly_smiling_face: